examples/ListDir.hs

```diff
@@ -23,5 +25,9 @@ main :: IO ()
 main = do
     hSetBuffering stdout LineBuffering
     let start = Stream.fromPure (Left ".")
-    Stream.iterateMapLeftsWith Stream.ahead listDir start
-        & Stream.mapM_ print
+    Stream.iterateMapLeftsWith ahead listDir start
+        & Stream.fold (Fold.drainMapM print)
+
+    where
+
+    ahead s1 s2 = Concur.ahead [s1, s2]
```

Comment:
This will have very bad performance. We need iterateMapWith in the Concurrent module with proper implementation using a single channel.